### ISSUE TITLE
gap in result images

### ISSUE THREAD

**Issue by lilydjwg** [naelstrof/maim#111]
![gap](https://user-images.githubusercontent.com/440661/29205068-30523ea2-7ead-11e7-8a62-f7cf9a4c5feb.png)

There is a vertical gap. The command is `maim -s -l -c 255,0,255,0.15 -k -n 2  > gap.png` and versions:

* Arch Linux
* maim v5.4.66
* Awesome WM 3.5.9
* xorg-server 1.19.3
* xcompmgr v1.1.7

**Comment by naelstrof** [comment]
Thanks, that's the one WM I didn't test after adding XShape compliance.
I'm possibly reconstructing the shape wrong, or Awesome WM is reporting the shape wrong. I'll see if I can replicate the issue tomorrow.

**Comment by vaygr** [comment]
I'm experiencing similar issue with maim **5.4.67**.

But for a 2560 x 1440 image, the "gap" is actually bigger and looks different:

![shot](https://user-images.githubusercontent.com/8810597/29288850-62fd438a-8108-11e7-9414-2f725675e8d1.png)

And that rectangle of transparency appears exactly in the same place for every shot.

Version **3.4.47** worked without this defect.

**Comment by lilydjwg** [comment]
I get only a thin cross with my Retina screen (192 dpi), other parts are transparent.

**Comment by naelstrof** [comment]
Could you guys try this tag? https://github.com/naelstrof/maim/releases/tag/v5.4.68

I couldn't replicate your issues, so I'm assuming there's some application that is a child of a window you're screenshotting that has some kind of XShape boundary/clipping stuff engaged.
I was improperly checking for bounding region changes, so that's gone. I also now ignore clipping regions when the root window is selected.

**Comment by lilydjwg** [comment]
That tag works fine for me, thanks 👍

**Comment by naelstrof** [comment]
I'll close the issue when @vaygr confirms it's fixed.

**Comment by vaygr** [comment]
I can confirm, **5.4.68** works perfectly well.

Just a quick non-related note: we should probably update the manpage as `maim image/path.png` no longer works, but indeed, `maim > image/path.png` does.

**Comment by naelstrof** [comment]
That's incorrect, `maim image/path.png` does work and is intended to work. Does it not work for you?

**Comment by vaygr** [comment]
Oh, so there's something wrong then.

Yes, `maim shot.png` just throws the screenshot to stdout.

**Comment by naelstrof** [comment]
Are you compiling with or without unicode support?
I might have to submit a bug report to cxxopts.

**Comment by vaygr** [comment]
Tried both w/ and w/o unicode support compiled in.

Yes, @naelstrof, your `--` suggestion worked, thanks. I was trying to quick-debug it in `parse()`, and `argv[current]` for positional argument couldn't fall into any of the conditions which would do something meaningful to that value and store it in options.

**Comment by vaygr** [comment]
That, however, ignored slop usage (and I suspect any other parameter) if set as the first argument. So it must always be specified right before the screenshot path. This way, where arguments order matters, it's kinda tricky.

**Comment by naelstrof** [comment]
I'd like to find a way to replicate it so I can try to fix it. Could you tell me your shell, terminal emulator, and  compiler (with version)?

**Comment by vaygr** [comment]
I should've finished my investigation on that. Sorry for the false alarm @naelstrof 

It seems the issue was my patch that replaces `std::regex` with [boost::regex](http://www.boost.org/doc/libs/1_57_0/libs/regex/doc/html/boost_regex/ref.html) in order to make `maim` portable and compilable with GCC 4.6 (as previous version did).

Turns out, while both `std::regex_match` and `boost::regex_match` return `true` or `false` in case of positive or negative match respectively, the effect of the latter on the result object is undefined in case of a no match, therefore `result.empty()` never returned `true`. I swapped those checks with function returns and that fixed the bug.

I apologize for the inconvenience, and thanks again for fixing initial issue that quick. I think it's safe to close it now.
